faq.md

```diff
@@ -499,6 +499,90 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
 profile or release mode using the menu entries **Run > Flutter Run in Profile Mode**
 or **Release Mode**.
 
+### What programming paradigm does Flutter's framework use?
+
+Flutter is a multi-paradigm programming environment. Many programming techniques
+developed over the past few decades are used in Flutter, each to solve
+a particular set of problems in a clean way. In no particular order:
+
+* Composition: The primary paradigm used by Flutter is that of using small objects with narrow
+scopes of behavior, composed together to obtain more complicated effects. Most widgets
+in the Flutter widget library are built in this way.
+
+* Functional programming: Entire applications can be built with only
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html)s,
+which are essentially functions that describe how arguments map to other functions,
+bottoming out in primitives that compute layouts or paint graphics.
+Additionally, heavy use is made of immutable data structures, including the entire
+[Widget](https://docs.flutter.io/flutter/widgets/Widget-class.html)
+class hierarchy as well as numerous supporting classes such as
+[Rect](https://docs.flutter.io/flutter/dart-ui/Rect-class.html) and
+[TextStyle](https://docs.flutter.io/flutter/painting/TextStyle-class.html).
+On a smaller scale, Dart's
+[Iterable](https://docs.flutter.io/flutter/dart-core/Iterable-class.html) API,
+which makes heavy use of the functional style (map, reduce, where, etc), is
+frequently used to process lists of values in the framework.
```

Comment:
What is the problem being solved? What is the benefit? Why do you use immutable data structures? The use of immutable data structures is not the same as functional programming (use of the former does not imply the latter, although the latter does require the former). Are you conflating the two?

Reply:
As above, I think we should avoid going into details of "why" here since it would lead to subjective arguments which isn't the point of the FAQ entry. The entry is just trying to answer the question "what paradigms does Flutter use", the answer to "why" could result in many book deals down the road. :-)